compiler/rustc_middle/src/ty/util.rs

```diff
@@ -845,8 +847,13 @@ impl<'tcx> ty::TyS<'tcx> {
             return (0, 0);
         }
 
-        let src_field_iter = self.ty_adt_def().unwrap().all_fields();
-        let mut bitmap: u64 = u64::MAX;
+        if adt_def.is_none() {
+            // Non-ADT types are not yet supported for precise tracing, so
+            // ensures they are scanned conservatively.
```

Comment:
s/ensures/ensure/

Should this be `FIXME` or even `todo!()`?

Reply:
A `todo` would prevent the compiler from compiling. I can make it a FIXME, but all this is saying is "this particular code path hasn't been optimised" - it doesn't change correctness.

Reply:
OK, `FIXME` sounds sensible.

Reply:
Fixed: 6195cc9396f35606216824f35185110670e71373